x-pack/plugins/infra/public/pages/metrics/inventory_view/components/layout.tsx

```diff
@@ -72,6 +75,12 @@ export const Layout = React.memo(
     const legendSteps = legend?.steps ?? DEFAULT_LEGEND.steps;
     const legendReverseColors = legend?.reverseColors ?? DEFAULT_LEGEND.reverseColors;
 
+    const [hostsLinkClicked, setHostsLinkClicked] = useLocalStorage<boolean>(
+      HOSTS_LINK_LOCAL_STORAGE_KEY,
+      false
+    );
+    const hostsLinkClickedRef = useRef<boolean | undefined>(hostsLinkClicked);
```

Comment:
**question**: why the need to store the boolean into a ref? `hostsLinkClicked` is already store into a react state and synced with the localStorage, that state should be enough for restoring the store value and conditionally render the button.

Reply:
Because we don't want the link to "disappear" the moment the user clicks on it. The idea is that the user is navigated to the hosts view and the button stays there after the click but when the user comes back to the infra view the button shouldn't be there anymore. We have the similar logic for the [KubernetesButton](https://github.com/jennypavlova/kibana/blob/150626-logs-ui-the-live-streaming-mode-uses-too-long-default-refresh-interval-from-the-time-service/x-pack/plugins/infra/public/pages/metrics/inventory_view/components/bottom_drawer.tsx/#L37)